Complete home work for 4 (fourth) sprint

Danzl opened the pull request.

Danzl committed:
Complete home work for 4 (fourth) sprint

ABashkirova commented on MovieQuiz-ios-project_sprint_3_start-3/MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -0,0 +1,227 @@
+    @IBOutlet private var noButton: UIButton!
+    
+    
+    
```
```suggestion

```

ABashkirova commented on MovieQuiz-ios-project_sprint_3_start-3/MovieQuiz/Helpers/UIColor+Extensions.swift:
```diff
@@ -0,0 +1,11 @@
+    static var green: UIColor? { UIColor(named: "green") }
+    static var gray: UIColor? { UIColor(named: "gray") }
+    static var red: UIColor? { UIColor(named: "red") }
+    static var background: UIColor? { UIColor(named: "background") }
```
не самый лучший вариант назвать переменные, так как у UIColor уже есть список констант black/white/ect 

Когда добавляешь свои цвета, надо определять некоторый namespace чтобы не было пересечений, например ypBlack. 

Все это нужно для того, чтобы избежать двусмысленности. Для меня UIColor.black - черный, системный. Когда я увижу это в коде, ожидаю что будет системный цвет работать

ABashkirova commented on MovieQuiz-ios-project_sprint_3_start-3/MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -0,0 +1,227 @@
+        }
+    }
+    
+    
```
```suggestion

```

ABashkirova commented on MovieQuiz-ios-project_sprint_3_start-3/MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -0,0 +1,227 @@
+    }
+    
+    
+    
```
```suggestion
    
```

ABashkirova commented on MovieQuiz-ios-project_sprint_3_start-3/MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -0,0 +1,227 @@
+        if isCorrect {
+            correctAnswer += 1
+        }
+        imageView.layer.masksToBounds = true
```
masksToBounds, cornerRadius - свойства которые можно настроить один раз

ABashkirova reviewed (changes requested):
1. При старте приложения у постера нет скруглений
<img width="515" alt="Снимок экрана 2022-11-14 в 17 52 56" src="https://user-images.githubusercontent.com/20269264/201678114-2a255f8a-e0d6-4724-a48c-6781fe7a3a0f.png">

2. Шрифты в проект не добавлены и при запуске не отображаются
3. По учебнику все сверстано очень точно. Однако по макету кнопки должны быть прижаты к низу экрана. Нужно добавить констрейнт главному стеку к safe area, не смотря на то что в учебнике этот пункт пропущен. Это действие может привести к тому, что текст вопроса перестанет помещаться в контейнере – здесь нужно пойти на компромис и добавить пространства (уменьшить констрейнты)
4. у launch screen картинки стоит неверный content mode, нужно поставить такой, чтобы изображение заполняло весь доступный размер

Danzl committed:
Добавлены шрифты, округления, констрейт, удалены пустые строки

1. При старте приложения у постера (Preview Image) имеется округление в 20 пунктов.
2.Добавлены шрифты в info.plist, так же они добавлены как сам файл в папке Fonts (MovieQuiz-ios-project_sprint_3_start-3/MovieQuiz/Resources/Assets/Fonts).
3.Добвлен констрейнт к главному стеку к safe area к низу экрана.
4. У launch screen картинки изменён content mode на Aspect Fill - изображение теперь заполняет весь размер экрана.
5. Удалены пустые строки.

ABashkirova commented:
1. Отступ от safe area должен быть нулевым. В фигме можно зажать cmd чтобы посомтреть расстояния:
<img width="375" alt="Снимок экрана 2022-11-15 в 21 33 05" src="https://user-images.githubusercontent.com/20269264/201987403-27866747-28c2-4e42-95f3-12c190692d1c.png"><img width="375" alt="Снимок экрана 2022-11-15 в 21 31 37" src="https://user-images.githubusercontent.com/20269264/201987430-0e83136b-ed3f-4538-82e9-38e1869e192c.png">

2. Шрифты добавленные в ассеты не подтянулись, к сожалению. Можешь переделать как показано в уроке?
3. Текст лейбла с вопросом не умещается :( Я предлагала выше, как это исправить. Предлагаю проверить запустив на других симуляторах.
@Danzl к сожалению не могу принять работу.

Danzl committed:
Шрифты, отступ от safe area, текст лейбла с вопросом 

1. Шрифты не слетают: надо было поставить target membership к MovieQuiz, чего не было показано/описано в уроке =(
2. Отступ от нижней части safe area нулевой: кнопки прижаты к низу экрана.
3. Изменены констрейты лейбла с вопросом (спасибо за совет), теперь он умещается на разных версиях.

Danzl commented:
1.
<img width="708" alt="Screenshot 2022-11-16 at 16 40 51" src="https://user-images.githubusercontent.com/115153894/202172657-7814d9f1-e82d-4ed8-96fa-0661444965c3.png">
2 + 3. 
<img width="783" alt="Screenshot 2022-11-16 at 16 49 03" src="https://user-images.githubusercontent.com/115153894/202173047-167626a1-287b-4c30-a13e-4de60c372a7e.png">

ABashkirova commented:
@Danzl мне очень жаль, что в ученбнике неверный скриншот. Спасибо большое за фидбек, я передам это команде авторов.

Вот в этом окне должна стоять галка включить в таргет проекта:
<img width="570" alt="Снимок экрана 2022-11-16 в 23 46 57" src="https://user-images.githubusercontent.com/20269264/202279371-e3ed59a4-cbe4-46a4-963c-102ccf40fbb8.png">

ABashkirova reviewed (changes requested):
исправления приняты!

ABashkirova commented:
Можно смерджить
